app/models/host/managed.rb

```diff
@@ -551,11 +556,33 @@ def jumpstart?
     operatingsystem.family == "Solaris" and architecture.name =~/Sparc/i rescue false
   end
 
+  def hostgroup_inherited_attributes
+    %w{puppet_proxy_id puppet_ca_proxy_id environment_id compute_profile_id}
+  end
+
+  def apply_inherited_attributes(attributes, initialized = true)
+    return nil unless attributes
+    new_hostgroup_id = attributes['hostgroup_id']
+    #hostgroup didn't change, no inheritance needs update.
+    return attributes if new_hostgroup_id.blank?
+
+    hostgroup = initialized && self.hostgroup ? self.hostgroup : Hostgroup.find_by_id(new_hostgroup_id)
```

Comment:
If there was no hostgroup set previously, or the hostgroup will be changed when the new attributes hash will be applied.